src/main/java/core/basesyntax/ConsoleHelper.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class ConsoleHelper {
+    private static final String YES_NO_HINT =
+            "(press 'y' for \"yes\" answer, other key for no)";
+    private static final String CORRECT_YES_ANSWER = "y";
+    private static ConsoleHelper instance;
+
+    public static ConsoleHelper getInstance() {
+        if (instance == null) {
+            instance = new ConsoleHelper();
+        }
+        return instance;
+    }
```

Comment:
create separate Factory class which will give instance of ConsoleHelper and FileHelper

Reply:
I try to implement Factory pattern for creating helpers and use it in tests for isolation MainManu class from console input and file output. So ConsoleHelper become a stub for input, and FileHelper insted of doing operations with files just checks if a correct method of FileHelper was invoked.
But I can't generate more informative names for Factories than RealFactory in main app and MockFactory in tests. In main app it looks confusing as for me, but I don't know what to do with it.